libraries/AP_HAL_Linux/GPIO_Sysfs.cpp

```diff
@@ -25,6 +25,18 @@ using namespace Linux;
 
 static const AP_HAL::HAL& hal = AP_HAL::get_HAL();
 
+#define UINT32_MAX_STR "4294967295"
+
+/* Trick to use minimum stack space for each of the params */
+union gpio_params {
+    char export_gpio[sizeof("export")];
+    char direction[sizeof("gpio" UINT32_MAX_STR "/direction")];
+    char value[sizeof("gpio" UINT32_MAX_STR "/value")];
+};
+
+#define GPIO_BASE_PATH "/sys/class/gpio/"
+#define GPIO_PATH_MAX (sizeof(GPIO_BASE_PATH) + sizeof(gpio_params) - 1)
+
```

Comment:
I think this method, which has been used elsewhere is a bit heavy for what it's doing and maybe allocating 256 bytes on the stack would be small enough and would be suitable for all cases. But It is no big deal.